lib/p2p/Pool.ts

```diff
@@ -85,20 +100,32 @@ class Pool extends EventEmitter {
     this.connected = false;
   }
 
+  /** Attempt to create an outbound connection to a node using its known listening addresses. */
+  private connectNode = async ({ addresses, nodePubKey }: NodeFactory) => {
+    for (let n = 0; n < addresses.length; n += 1) {
+      try {
+        await this.addOutbound(addresses[n], nodePubKey);
```

Comment:
why connecting to all the node's addresses? if one succeed, isn't it sufficient? specially if they are all using the same pubkey.